### DISCUSSION TITLE
Get demo analysis progress?

### DISCUSSION THREAD

**Discussion by justMemou** [markus-wa/demoinfocs-golang#568] [General]
Heya,

As parsing header doesn't seem to work - is there any way to determine what % of the demo is being currently analyzed?
I'm trying the following, but doesn't seem to work, as the PlaybackFrames seems to be 0 on a standard faceit demo, maybe I'm doing it wrong?:
```GOLANG
	header, err := p.ParseHeader()
	if err != nil {
		panic(err)
	}
	totalFrames := header.PlaybackFrames

	p.RegisterEventHandler(func(e events.FrameDone) {
		currentFrame := p.CurrentFrame()
		progress := float64(currentFrame) / float64(totalFrames) * 100
		fmt.Printf("Analysis Progress: %.2f%% (Frame %d of %d)\n", progress, currentFrame, totalFrames)
	})
```

**Comment by markus-wa** [comment]
I think the docs need updating to deprecate `ParseHeader` for CS2

try

```go
p.RegisterNetMessageHandler(func(h *msgs2.CDemoFileHeader) {
...
})
```

**Reply by justMemou** [reply]
unfortunately doesn't seem to contain such data:
`CDemoFileHeader:  demo_file_stamp:"PBDEMS2\x00" network_protocol:14030 server_name:"FACEIT.com register to play here" client_name:"SourceTV Demo" map_name:"de_mirage" game_directory:"/home/******/cs2/game/csgo" fullpackets_version:2 allow_clientside_entities:true allow_clientside_particles:true addons:"" demo_version_name:"valve_demo_2" demo_version_guid:"******-04a1-4c01-bb61-*******" build_num:10072 server_start_tick:4624`
